### PR TITLE
Created pipelines for the External CI project

### PR DESCRIPTION
**Pull request by abhimeda**
Created pipelines for the following components. 

- AMDMIGraphX
- hipfort
- rocSOLVER
- rocWMMA

Co-authored-by: amd-jmacaran [joseph.macaranas@amd.com](mailto:joseph.macaranas@amd.com)
Co-authored-by: alexxu-amd [alex.xu@amd.com](mailto:alex.xu@amd.com)

P.S. Apologies for the branch name. I realized I could batch all of these into one PR instead of four and was hesitant to rename the branch in case it would break something.

### CODE REVIEW

File: .azuredevops/components/rocSOLVER.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - rocsparse
+    - hipsparse
+    - rocsolver
+    - libsuitesparse-dev
+    - gfortran
+    - libfmt-dev
+    - git
+jobs:
+- job: rocSOLVER
+  timeoutInMinutes: 180
```
Comment (amd-jmacaran) [inline .azuredevops/components/rocSOLVER.yml:23]: Remove this timeout

File: .azuredevops/components/rocSOLVER.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - cmake
+    - ninja-build
+    - rocblas
+    - rocsparse
+    - hipsparse
+    - rocsolver
+    - libsuitesparse-dev
+    - gfortran
+    - libfmt-dev
+    - git
```
Comment (amd-jmacaran) [inline .azuredevops/components/rocSOLVER.yml:20]: Add a new line in between the blocks here

File: .azuredevops/components/rocWMMA.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  default:
+    - cmake
+    - ninja-build
+    - libboost-program-options-dev
+    - libgtest-dev
+    - googletest
+    - libfftw3-dev
+    - git
+    - rocblas
+jobs:
```
Comment (amd-jmacaran) [inline .azuredevops/components/rocWMMA.yml:19]: Add new-line in between here

**Review by amd-jmacaran — CHANGES_REQUESTED** [review]

File: .azuredevops/components/rocSOLVER.yml (hunk 1 above)
Comment (abhimeda) [reply .azuredevops/components/rocSOLVER.yml]: done

File: .azuredevops/components/rocSOLVER.yml (hunk 2 above)
Comment (abhimeda) [reply .azuredevops/components/rocSOLVER.yml:20]: done

File: .azuredevops/components/AMDMIGraphX.yml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  - script: 'ls -1R /opt/rocm'
+    displayName: 'Artifact listing'
+  - script: 'ls -1R $(Agent.BuildDirectory)/rocm'
+    displayName: 'Artifact listing'
+  - template: ${{ variables.CI_TEMPLATE_PATH }}/steps/build-cmake.yml
+    parameters:
+      extraBuildFlags: >-
+        -DCMAKE_CXX_COMPILER=/opt/rocm/llvm/bin/clang++
+        -DCMAKE_C_COMPILER=/opt/rocm/llvm/bin/clang
+        -DCMAKE_PREFIX_PATH=/opt/rocm
```
Comment (alexxu-amd) [inline .azuredevops/components/AMDMIGraphX.yml:99]: -DCMAKE_PREFIX_PATH is defined twice

**Review by alexxu-amd — CHANGES_REQUESTED** [review]

File: .azuredevops/components/AMDMIGraphX.yml (hunk 4 above)
Comment (abhimeda) [reply .azuredevops/components/AMDMIGraphX.yml]: fixed

**Review by alexxu-amd — DISMISSED** [review]

**Review by amd-jmacaran — APPROVED** [review]

**Review by alexxu-amd — APPROVED** [review]

**Pull request merged**